R/verb-slice.R

```diff
@@ -131,9 +131,10 @@ slice_by <- function(.data, order_by, size, .by, with_ties = FALSE) {
     )
   }
 
-  out <- .data %>%
-    window_order({{order_by}}) %>%
-    filter(!!window_fun) %>%
+  # must use `add_order()` as `window_order()` only allows variables
+  .data$lazy_query <- add_order(.data, quos({{order_by}}))
```

Comment:
Does any existing test cover this? What happens if there's a select after this step?

Reply:
The previous window order is restored below and there is a test for this. A `select()` after this step would drop the window order in case of `slice_sample()` and keep the window order for the others (unless the window order variables are de-selected).